### ISSUE TITLE
Question: scaling the ocean to match 1:1 scaling and not 1:50/1:100

### ISSUE THREAD

**Issue by gameanimation3d** [huwb/crest-oceanrender#129]
Hey,
Is it possible to scale the ocean so it matches an 1m = 1 Unity Unit scaling and not 1:50 or 1:100 as its currently.
I tried already to scale the waves(e.g gerstner waves) down but i didn't found a Solution.
Because i currently scale everything else bigger to match the ocean scaling. But only scaling the ocean would be better because e.g light backing for Objects takes way longer in such a scaling.

**Comment by huwb** [comment]
Hi, I'm not sure i understand the question. Everything should be to scale, in that the wavelengths in the gerstner wave spectrum should be world units - so the ocean should be 1:1 already. The scale of the geometry is inconsequential and the scaling on the Ocean transform is usually left to be completely automatic, and has no effect on the 'world scale' in terms of the wave sizes etc. If you have a specific issue that you can illustrate maybe i can comment further.

**Comment by huwb** [comment]
To test this you can set the gerstner waves script to generate just 1 component per octave (instead of 5 or whatever you have it set to), and then enable only one octave in the spectrum, such as the 2m, and check the resulting wave is on the order of 2m (it should be between 2 and 4m depending on RNG)

**Comment by gameanimation3d** [comment]
Hey, 
sorry that i respond to this topic so late but we had some serious milestone for my animation project in the university which we use crest for.
![grafik](https://user-images.githubusercontent.com/44543590/50217916-7a13d100-038a-11e9-8f69-42981d90dee6.png)
For illustration i added an image. 
if scale everything to 1 Unity Units  = 1m the tile distribution and amount doesn't adjust to them.
So i only get 4 Tiles compared roughly 20-30 LOD Tiles before.
And for the Gerstner Wave they are now really big. I tried your adjustments

> To test this you can set the gerstner waves script to generate just 1 component per octave (instead of 5 or whatever you have it set to), and then enable only one octave in the spectrum, such as the 2m, and check the resulting wave is on the order of 2m (it should be between 2 and 4m depending on RNG)

But the waves still remain the very big. How can i scale them down?

Btw. i use Crest 4.0 and not the current master.

**Comment by huwb** [comment]
Unfortunately i still don't understand, i think  there is some confusion here. Let me try a different angle by illustrating a concrete example:

If i wanted waves that have a wavelength of lets say ~2m, i would do precisely this:

* I would do *nothing* with the scales on any ocean game object's transform. I would ignore the Scale set on the transforms completely, because its not relevant to the size of the waves. Changing the scale should never be necessary.
* I would create a wave spectrum and turn all the sliders down.
* I would turn up the slider for "2" to get ~2m waves

So basically waves should have the correct wavelength, and you should never need to change any scales of anything manually.

**Comment by gameanimation3d** [comment]
Hey,
thank you for the answer. i think didn't explained the problem very well, sorry.
if have all assets scaled to to an scaling 1m = 2 unity units(1:2).
On such a "real world" scaling  the oceans subdivision is very low. Dynamic Waves aren't possible with resolution.
![unbenannt](https://user-images.githubusercontent.com/44543590/50696004-98422800-103e-11e9-8c2c-a4ab0463e6e4.JPG)


Therefore i usually use an scaling of 1:100. To get a decent subdivision
![unbenannt2](https://user-images.githubusercontent.com/44543590/50696069-c162b880-103e-11e9-99b2-422958afa986.JPG)
The Ocean Settings:
![grafik](https://user-images.githubusercontent.com/44543590/50696178-01c23680-103f-11e9-9ec0-9f2b3a84fab0.png)

It seem for me the LOD System doesn't adapt how close/small the camera is in the world. Cranking up the settings even more doesn't help as already many unseen polygons are in the scene .
The distance between the duck and the camera is on the images relativly the same. Only the icons are smaller due to the scaling.
I leave the ocean completly untouched. I scale only the asset with an overall empty gameobject.


> If i wanted waves that have a wavelength of lets say ~2m, i would do precisely this:

    I would do nothing with the scales on any ocean game object's transform. I would ignore the Scale set on the transforms completely, because its not relevant to the size of the waves. Changing the scale should never be necessary.
    I would create a wave spectrum and turn all the sliders down.
    I would turn up the slider for "2" to get ~2m waves
>

i tried your solution this works for the scaling of the gerstner waves.

Another issue in a scaling of 1:2:
The normal mapping in the ocean shader now seems to be moving relativly fast. scaling the map very big only helps. So small/mid-waves from the normal map aren't possible without fast moving/ texture sliding. Can i upload a video here on github to visualize it?

I am still using the  Crest 4.0 Release Version. Or does Crest 5 solves this problem?

**Comment by huwb** [comment]
No you didnt explain any of that :).

I highly unrecommend having an arbitrary scale on your assets. Crest assumes 1m scale, other things in unity probably do the same.

You can find the normal map scrolling in oceannormal.shader. you can find where it depends on time and speed it up. I'll contemplate promoting it to a material Param. If I do I'll let you know.

That's the best I can do in my free time, apologies for not being able to be more helpful.
